### PR TITLE
MINOR: [Docs] Fix sphinx rst syntax in Integration page

### PR DESCRIPTION
**Pull request by jorisvandenbossche**

### CODE REVIEW

**Review by lidavidm — APPROVED** [review]

File: docs/source/format/Integration.rst [hunk 1]
```diff
@@ -455,14 +455,16 @@ Gold File Integration Tests
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Pre-generated json and arrow IPC files (both file and stream format) exist
-in the `arrow-testing <https://github.com/apache/arrow-testing>` repository
+in the `arrow-testing <https://github.com/apache/arrow-testing>`__ repository
```
Comment (kou) [inline docs/source/format/Integration.rst:458]: Do we need two `_`s here?

```suggestion
in the `arrow-testing <https://github.com/apache/arrow-testing>`_ repository
```

Comment (jorisvandenbossche) [reply docs/source/format/Integration.rst:458]: Strictly speaking the double underscore is "more" correct if this link name (arrow-testing) isn't reused, so I have the habit of writing the double underscore for links like this (https://stackoverflow.com/questions/27420317/restructured-text-rst-http-links-underscore-vs-use). But in practice it doesn't matter really, both work fine in most cases.

Comment (kou) [reply docs/source/format/Integration.rst:458]: Wow! I didn't know it. Thanks!

**Review by kou — APPROVED** [review]
+1

**Pull request merged**